homeassistant/components/maxcube/climate.py

```diff
@@ -151,37 +131,44 @@ def hvac_modes(self):
 
     def set_hvac_mode(self, hvac_mode: str):
         """Set new target hvac mode."""
-        device = self._cubehandle.cube.device_by_rf(self._rf_address)
-        temp = device.target_temperature
-        mode = MAX_DEVICE_MODE_MANUAL
-
         if hvac_mode == HVAC_MODE_OFF:
-            temp = OFF_TEMPERATURE
+            self._set_target(MAX_DEVICE_MODE_MANUAL, OFF_TEMPERATURE)
         elif hvac_mode == HVAC_MODE_HEAT:
-            temp = max(temp, self.min_temp)
-        else:
-            temp = None
-            mode = MAX_DEVICE_MODE_AUTOMATIC
-
-        cube = self._cubehandle.cube
+            temp = max(self._device.target_temperature, self.min_temp)
+            self._set_target(MAX_DEVICE_MODE_MANUAL, temp)
+        elif hvac_mode == HVAC_MODE_AUTO:
+            self._set_target(MAX_DEVICE_MODE_AUTOMATIC, None)
```

Comment:
Would be good to raise a ValueError if an hvac_mode is passed that isn't supported. Ie handle the `else:` case.

Reply:
Done